### PR TITLE
8275082: Update XML Security for Java to 2.3.0

### PR DESCRIPTION
**Pull request by wangweij**
Import Apache Santuario 2.3.0 without the secure validation changes since in OpenJDK we are using the `jdk.xml.dsig.secureValidationPolicy` security property for XML Signature secure validation protection.

Two commits are pushed:

- 2.3.0: Import 2.3.0 code changes
- revert: revert the Santuario secure validation changes

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8275082](https://bugs.openjdk.java.net/browse/JDK-8275082): Update XML Security for Java to 2.3.0

### Reviewers
 * [Sean Mullan](https://openjdk.java.net/census#mullan) (@seanjmullan - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/6644/head:pull/6644` \
`$ git checkout pull/6644`

Update a local copy of the PR: \
`$ git checkout pull/6644` \
`$ git pull https://git.openjdk.java.net/jdk pull/6644/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 6644`

View PR using the GUI difftool: \
`$ git pr show -t 6644`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/6644.diff">https://git.openjdk.java.net/jdk/pull/6644.diff</a>

</details>

### CODE REVIEW

**Comment by valeriepeng** [comment]
The revert is done manually and covered by existing regression tests? May need to add a noreg-xxx label

File: src/java.xml.crypto/share/classes/org/jcp/xml/dsig/internal/dom/Policy.java [hunk 1]
```diff
@@ -44,13 +44,13 @@
...
-    private static Map<String, Integer> minKeyMap = new HashMap<>();
-    private static boolean noDuplicateIds = false;
-    private static boolean noRMLoops = false;
+    private static Set<URI> disallowedAlgs;
+    private static int maxTrans;
+    private static int maxRefs;
+    private static Set<String> disallowedRefUriSchemes;
+    private static Map<String, Integer> minKeyMap;
+    private static boolean noDuplicateIds;
+    private static boolean noRMLoops;
```
Comment (valeriepeng) [inline src/java.xml.crypto/share/classes/org/jcp/xml/dsig/internal/dom/Policy.java:53]: Does it really make any difference, moving the initialization to the initialize() method? The comment on line 46 no longer apply now that the initialization is relocated.

Comment (wangweij) [reply src/java.xml.crypto/share/classes/org/jcp/xml/dsig/internal/dom/Policy.java:53]: No for good people. I was writing some local tests to add and remove rules in an arbitrary style, and cleaning up inside the `initialize()` method makes this possible, albeit still a little ugly (need `--add-opens`).

I'll remove that comment.

**Comment by wangweij** [comment]
> The revert is done manually and covered by existing regression tests? May need to add a noreg-xxx label

Yes. I'll find a proper label.

**Review by seanjmullan — APPROVED** [review]

**Pull request closed without merging**